pkg/sql/schemachanger/scbuild/internal/scbuildstmt/create_index.go

```diff
@@ -590,7 +670,16 @@ func createVirtualColumnForIndex(
 			panic(err)
 		}
 		d.Type = typedExpr.ResolvedType()
+		validateColumnIndexableType(typedExpr.ResolvedType())
 	}
 	alterTableAddColumn(b, tn, tbl, &tree.AlterTableAddColumn{ColumnDef: d})
+	// Mutate the accessibility flag on this column it should be inaccessible.
+	{
+		ers := b.ResolveColumn(tbl.TableID, d.Name, ResolveParams{
+			RequiredPrivilege: privilege.CREATE,
+		})
+		_, _, col := scpb.FindColumn(ers)
+		col.IsInaccessible = true
+	}
```

Comment:
Why? Do you mean a virtual column is always inaccessible?

Reply:
Yes, any new virtual columns added by CREATE INDEX need to be inaccessible, this is the current behaviour. Just being hidden isn't sufficient.

Reply:
I'll add a comment to make it more clear:

// When a virtual column for an index expression gets added for CREATE INDEX it must be inaccessible,
// so we will manipulate the newly added element to be inaccessible after going through the normal add
// column code path.